flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/RowDataToCsvConverters.java

```diff
@@ -128,7 +128,7 @@ private static RowFieldConverter createRowFieldConverter(LogicalType fieldType)
 			case DATE:
 				return (csvMapper, container, row, pos) -> convertDate(row.getInt(pos), container);
 			case TIME_WITHOUT_TIME_ZONE:
-				return (csvMapper, container, row, pos) -> convertTime(row.getInt(pos), container);
+				return (csvMapper, container, row, pos) -> convertTime(row.getLong(pos), container);
```

Comment:
We should still use `getInt`, because we always use `int` to represent number of milliseconds of the day in Flink SQL internally. See Javadoc of `RowData`.

Reply:
Will overflow when supporting nanosecond precision in the future

Reply:
That's why we don't support `precision > 3` for now.
When the planner supports `precision > 3` in the future, we can update csv format to use the higer precision data structure (maybe long or otheres).

Reply:
get it